app/javascript/psulib_blacklight/javascripts/availability.js

```diff
@@ -228,50 +257,73 @@ function groupByLibrary(holdings) {
     }, {});
 }
 
-function resolveStatus(chargeable, homeLocationID, currentLocationID, titleID) {
-    var statusCode = '';
-    var statusText = '';
-
-    if (chargeable === 'true') {
-        if (homeLocationID === 'ON-ORDER') {
-            statusCode = 'ON-ORDER';
-            statusText = 'Being Acquired by the Library';
-        } else {
-            statusCode = statusText = 'Available';
-        }
-    }
-    else {
-        switch (currentLocationID) {
-            case 'CHECKEDOUT':
-                statusCode = statusText = 'Checked Out';
-                break;
-            case 'MISSING':
-                statusCode = statusText = 'Missing';
-                break;
-            case 'ILLEND':
-                statusCode = 'ILLEND';
-                statusText = `<a data-type="ill-link" data-catkey="${titleID}" href="#"><span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span> This copy unavailable, submit request via Interlibrary Loan</a>`;
-                break;
-            default:
-                statusCode = statusText = 'Not Available';
-        }
+function resolveLocation(item, request_via_ill_locations, all_locations) {
+    var location = '';
+    var spinner = '<span class="spinner-border spinner-border-sm" role="status" aria-hidden="true"></span>';
+
+    // Check request via ILL locations
+    if (item.currentLocationID in request_via_ill_locations) {
+        location = `<a data-type="ill-link" data-catkey="${item.catkey}" data-call-number="${item.callNumber}" href="#">${spinner}Copy unavailable, request via Interlibrary Loan</a>`;
+    } else if (['ARKTHESES', 'AH-X-TRANS'].includes(item.currentLocationID)) {
+        var aeonLocation = (item.currentLocationID in all_locations) ? all_locations[item.currentLocationID] : "";
+        var shared = `data-catkey="${item.catkey}" data-call-number="${item.callNumber}" data-archival-thesis`;
+
+        location = `<a data-type="ill-link" ${shared} href="#">${spinner}Request Scan - Penn State Users</a><br>
+                    <a href="https://psu.illiad.oclc.org/illiad/upm/lending/lendinglogon.html">Request Scan - Guest</a><br>
+                    <a data-type="aeon-link" ${shared} data-item-id="${item.itemID}" data-item-location="${aeonLocation}" href="#">${spinner}View in Special Collections</a>`;
+    } else {
+        location = (item.currentLocationID in all_locations) ? all_locations[item.currentLocationID] : "";
```

Comment:
This pattern shows up a couple of times. Would be nice to have a generic function to say "map this string to the value of the key that matches that string on this object or return an empty string. Call it "mapLocation" then you could read the code easier and eliminate a nested if. Like

`location = mapLocation(currentLocationID, mapObject)`

Reply:
I meant this as a question

Reply:
yes, I was kinda thinking about doing the same thing but forget about it later on, will do it 👍

Reply:
updated